octobot_trading/data/order.py

```diff
@@ -153,43 +164,61 @@ def update(self, symbol, order_id="", status=OrderStatus.OPEN,
                 self.executed_time = self.exchange_manager.exchange.get_uniform_timestamp(timestamp)
         return changed
 
-    async def update_order_status(self, last_prices: list):
+    async def initialize_impl(self):
+        if not self.exchange_manager.is_backtesting:
+            asyncio.get_event_loop().call_later(self.CHECK_ORDER_STATUS_AFTER_INIT_DELAY,
+                                                asyncio.create_task,
+                                                self.update_order_status())
+        else:
+            asyncio.create_task(self.update_order_status())
+            await wait_asyncio_next_cycle()
+
+    async def update_order_status(self, force_refresh=False):
         """
         Update_order_status will define the rules for a simulated order to be filled / canceled
         """
         raise NotImplementedError("Update_order_status not implemented")
 
-    # check_last_prices is used to collect data to perform the order update_order_status process
-    def check_last_prices(self, last_prices, price_to_check, inferior) -> bool:
-        if last_prices:
-            prices = [p[ECOC.PRICE.value]
-                      for p in last_prices
-                      if not math.isnan(p[ECOC.PRICE.value]) and (
-                              p[ECOC.TIMESTAMP.value] >= self.creation_time)]
-
-            if prices:
-                if inferior:
-                    if float(min(prices)) < price_to_check:
-                        get_logger(self.get_name()).debug(f"{self.symbol} last prices: {prices}, "
-                                                          f"ask for {'inferior' if inferior else 'superior'} "
-                                                          f"to {price_to_check}")
-                        return True
-                else:
-                    if float(max(prices)) > price_to_check:
-                        get_logger(self.get_name()).debug(f"{self.symbol} last prices: {prices}, "
-                                                          f"ask for {'inferior' if inferior else 'superior'} "
-                                                          f"to {price_to_check}")
-                        return True
-        return False
-
     def cancel_order(self):
         """
         Set cancelled status and keep track of cancellation time.
-        :return: None
         """
         self.status = OrderStatus.CANCELED
         self.canceled_time = self.exchange_manager.exchange.get_exchange_current_time()
 
+    async def on_fill(self):
+        """
+        Set filled status
+        """
+        self.status = OrderStatus.FILLED
+        self.executed_time = self.generate_executed_time()
+
+    async def on_fill_complete(self):
```

Comment:
should this be called post_fill_action or something like that ? on_xxx is usually related to event oriented programming